### PR TITLE
Hard Light Mark 2 and keymaps

### PR DESCRIPTION
**Pull request by FateEverywhere**
This time with minor competence!

## Description

I have learned what was causing a bunch of weird conflicting changes with prior PRs, and will adjust for this in future. Thanks in advance for your patience.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/edi/hardlight/mk2/rules.mk [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+VELOCIKEY_ENABLE = yes
+
+# Layout definitions
+# LAYOUTS = ortho_4x16
+
+# Wildcard to allow APM32 MCU 
+DFU_SUFFIX_ARGS = -p FFFF -v FFFF
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```
Comment (fauxpark) [inline keyboards/edi/hardlight/mk2/rules.mk:45]:
```suggestion
# MCU name
MCU = STM32F072

# Bootloader selection
Bootloader = stm32-dfu

# Wildcard to allow APM32 MCU 
DFU_SUFFIX_ARGS = -v FFFF -p FFFF

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
VELOCIKEY_ENABLE = yes
KEY_LOCK_ENABLE = yes
WS2812_DRIVER = pwm

# Layout definitions
LAYOUTS = ortho_4x16

# Enter lower-power sleep mode when on the ChibiOS idle thread
OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```

File: keyboards/edi/hardlight/mk2/keymaps/kate/config.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/*
```
Comment (fauxpark) [inline keyboards/edi/hardlight/mk2/keymaps/kate/config.h:1]: This file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/edi/hardlight/mk2/keymaps/kate/config.h (hunk 2 above)
Comment (FateEverywhere) [reply keyboards/edi/hardlight/mk2/keymaps/kate/config.h]: Done!

File: keyboards/edi/hardlight/mk2/rules.mk [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/edi/hardlight/mk2/rules.mk:24]:
```suggestion
```

File: keyboards/edi/hardlight/mk2/rules.mk [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# Bootloader selection
+Bootloader = stm32-dfu
+
+# Wildcard to allow APM32 MCU 
+DFU_SUFFIX_ARGS = -v FFFF -p FFFF
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/edi/hardlight/mk2/rules.mk:13]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/edi/hardlight/mk2/rules.mk [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# MCU name
+MCU = STM32F072
+
+# Bootloader selection
+Bootloader = stm32-dfu
+
+# Wildcard to allow APM32 MCU
```
Comment (fauxpark) [inline keyboards/edi/hardlight/mk2/rules.mk:7]:
```suggestion
# Wildcard to allow APM32 MCU
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

File: keyboards/edi/hardlight/mk2/rules.mk [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+VELOCIKEY_ENABLE = yes
+KEY_LOCK_ENABLE = yes
+WS2812_DRIVER = pwm
+
+# Layout definitions
+LAYOUTS = ortho_4x16
```
Comment (noroadsleft) [inline keyboards/edi/hardlight/mk2/rules.mk:30]: We don't actually have this but I think I'll add one.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**